### PR TITLE
Delete old compatibility from the bigarray stubs

### PR DESCRIPTION
**Pull request by ghost**
Now that the `Bigarray.*.map_file` functions are gone, these macros are no longer necessary. This PR deletes them and substitute them in the C code.

### CODE REVIEW

**Review by xavierleroy — COMMENTED** [review]
I'm happy to get rid of this hack.  I forgot how ugly it was!  

I'll give it a round of CI precheck just to make sure.

File: otherlibs/win32unix/mmap.c [hunk 1]
```diff
@@ -26,31 +26,12 @@
...
-#define MAP_FILE_ERROR() caml_ba_sys_error()
-#else
-#define MAP_FILE_FUNCTION caml_unix_map_file
-#define MAP_FILE_FUNCTION_BYTECODE caml_unix_map_file_bytecode
-#define UNMAP_FILE_FUNCTION caml_unix_unmap_file
-#define ALLOC_FUNCTION caml_unix_mapped_alloc
-#define MAP_FILE_FUNCTION caml_unix_map_file
-#define CAML_MAP_FILE "Unix.map_file"
-#define MAP_FILE_ERROR() \
+#define uerror("map_file", Nothing) \
```
Comment (xavierleroy) [inline otherlibs/win32unix/mmap.c]: CI is unhappy, and rightly so:
```
cl -c -nologo -O2 -Gy- -MD   -I../unix -D_CRT_SECURE_NO_DEPRECATE -DCAML_NAME_SPACE -DUNICODE -D_UNICODE -DWINDOWS_UNICODE=1 -I../../runtime -Fommap.obj mmap.c
mmap.c
mmap.c(29) : error C2010: '"' : unexpected in macro formal parameter list

mmap.c:29:16: error: ""map_file"" may not appear in macro parameter list
 #define uerror("map_file", Nothing) \
                ^~~~~~~~~~

```

**Review by xavierleroy — CHANGES_REQUESTED** [review]

File: otherlibs/win32unix/mmap.c (hunk 1 above)
Comment (ghost) [reply otherlibs/win32unix/mmap.c]: Oops, it's fixed

**Comment by ghost** [comment]
I restarted precheck

**Review by xavierleroy — APPROVED** [review]
CI looks happy (precheck/220).  Will merge next.

**Pull request merged**